cppcache/integration-test/ThinClientSecurityHelper.hpp

```diff
@@ -198,18 +198,26 @@ class putThread : public ACE_Task_Base {
   }
 
   void start() {
-    m_run = true;
-    activate(THR_NEW_LWP | THR_JOINABLE, m_numthreads);
+    for(auto i = 0; i < m_numthreads; ++i) {
+      threads_.emplace_back([this](){
+        run();
+      });
+    }
+  }
+
+  void wait() {
+    for(auto& thread : threads_) {
+      if(thread.joinable()) {
+        thread.join();
+      }
+    }
   }
 
   void stop() {
-    if (m_run) {
-      m_run = false;
```

Comment:
With the removal of `m_run = false` what is left to tell the tasks to end?

Reply:
Thanks for pointing this out. Thing is there is no mechanism now to tell the task to end, as there wasn't before. Because m_run was not used, except as a mean to tell whether or not the thread was started, but that was not necessary as the thread was run in a controlled environment and it was not going to be stopped before it was started, and this last part is not different now.